src/udev/udevadm-verify.c

```diff
@@ -119,20 +119,45 @@ static int verify_rules_file(UdevRules *rules, const char *fname) {
         return 0;
 }
 
-static int verify_rules(UdevRules *rules, char **files) {
-        size_t fail_count = 0, success_count = 0;
+static int verify_rules_filelist(UdevRules *rules, char **files, size_t *fail_count, size_t *success_count);
+
+static int verify_rules_dir(UdevRules *rules, const char *dir, size_t *fail_count, size_t *success_count) {
+        int r;
+        _cleanup_strv_free_ char **files = NULL;
+
+        r = conf_files_list(&files, ".rules", NULL, 0, dir);
+        if (r < 0)
+                return log_error_errno(r, "Failed to enumerate rules files: %m");
+
+        return verify_rules_filelist(rules, files, fail_count, success_count);
+}
+
+static int verify_rules_filelist(UdevRules *rules, char **files, size_t *fail_count, size_t *success_count) {
         int r, rv = 0;
 
         STRV_FOREACH(fp, files) {
-                r = verify_rules_file(rules, *fp);
-                if (r < 0) {
-                        fail_count++;
-                        if (rv >= 0)
-                                rv = r;
-                } else
-                        success_count++;
+                if (endswith(*fp, "/"))
```

Comment:
I don't think this is very user friendly. The decision should be based on whether the path is really a directory, and not based on how the user specified it. (I know I wrote something like this in telegram, but that was just a telegram shorthand… We don't have logic like this in other places in the codebase.)

Reply:
OK, let's just descent into directories.

Reply:
But in that case extra care should be taken to avoid directory loops.